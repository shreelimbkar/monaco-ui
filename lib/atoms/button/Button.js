"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RRSButton = void 0;
var react_1 = __importDefault(require("react"));
// import styled, { AnyStyledComponent, css } from "styled-components";
var ButtonStyle_1 = require("./ButtonStyle");
// interface IHeadingStyled {
//   primary?: boolean;
//   round?: boolean;
// }
var RRSButton = function (_a) {
    var onClick = _a.onClick, children = _a.children, props = __rest(_a, ["onClick", "children"]);
    return (react_1.default.createElement(ButtonStyle_1.ButtonStyles, __assign({}, props, { onClick: onClick }), children));
};
exports.RRSButton = RRSButton;
//# sourceMappingURL=Button.js.map