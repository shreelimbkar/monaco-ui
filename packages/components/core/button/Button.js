"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importDefault(require("react"));
var ButtonStyles_1 = require("./ButtonStyles");
var Button = function (props) {
    var onClick = props.onClick, children = props.children;
    return (react_1.default.createElement(ButtonStyles_1.ButtonStyles, __assign({}, props, { onClick: onClick }), children));
};
exports.default = Button;
