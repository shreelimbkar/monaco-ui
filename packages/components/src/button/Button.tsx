import React, { MouseEventHandler } from 'react';
import { ButtonStyles } from './ButtonStyles';

import DefaultProps from '../types/ButtonProps';

interface ButtonProps extends DefaultProps {
  children?: React.ReactNode;
  onClick?: MouseEventHandler;
  style?: object;
  className?: string;
}

const Button: React.FC<ButtonProps> = (props) => {
  const { onClick, children } = props;
  return (
    <ButtonStyles {...props} onClick={onClick}>
      {children}
    </ButtonStyles>
  );
};

export default Button;
