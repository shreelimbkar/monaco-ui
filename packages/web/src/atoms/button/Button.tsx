import React, { ButtonHTMLAttributes, MouseEventHandler } from "react";
// import styled, { AnyStyledComponent, css } from "styled-components";
import { ButtonStyles } from "./ButtonStyle";

interface DefaultProps {
  children?: React.ReactNode;
  props?: any;
  primary?: boolean;
  round?: boolean;
  disabled?: boolean;
  onClick?: MouseEventHandler;
  type?: "submit" | "reset" | "button";
}
// interface IHeadingStyled {
//   primary?: boolean;
//   round?: boolean;
// }

const RRSButton: React.FC<DefaultProps> = ({ onClick, children, ...props }) => {
  return (
    <ButtonStyles {...props} onClick={onClick}>
      {children}
    </ButtonStyles>
  );
};

// const ButtonStyles = styled.button<IHeadingStyled>`
//   background: black;
//   color: white;
//   border-radius: 7px;
//   padding: 20px;
//   margin: 10px;
//   font-size: 16px;
//   :disabled {
//     opacity: 0.4;
//   }
//   :hover {
//     box-shadow: 0 0 10px yellow;
//   }
//   ${(props) =>
//     props.primary &&
//     css`
//       background: green;
//       color: white;
//     `}
//   border-radius: ${(props) => (props.round ? "50%" : "7px")}
// `;

export { RRSButton };
