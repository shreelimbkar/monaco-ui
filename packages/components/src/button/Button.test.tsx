import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Button from '.';

afterEach(cleanup);

describe('Button Component', () => {
  test('Renders Correctly', () => {
    const { asFragment } = render(<Button />);
    expect(asFragment()).toMatchSnapshot();
  });
});
