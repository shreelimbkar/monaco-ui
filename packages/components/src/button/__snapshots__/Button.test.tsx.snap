// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Renders Correctly 1`] = `
<DocumentFragment>
  <button
    class="sc-bdvvtL esTVur"
  />
</DocumentFragment>
`;
